Backend of app is done

letit6E opened the pull request:
Application backend implemented, streamlit added

letit6E committed:
Frontend of app add

letit6E committed:
Create README.md

letit6E committed:
Fix html block filtering

letit6E committed:
Cosmetic changes

SpirinEgor reviewed (changes requested):
Avoid global code for both the Streamlit interface and pipeline initialization.

letit6E committed:
Update main_summarizer.py

Eliminate global variables, move functionality to the class, fix summarization function

letit6E committed:
Update main.py

Move functionality to main function

letit6E committed:
Update main_summarizer.py

Move constants to class, make __init__ function of class

SpirinEgor commented on main.py:
```diff
@@ -0,0 +1,31 @@
+from syntax_summarizer import get_text
+
+if __name__ == '__main__':
+	cnn_summarizer = Summarizer()
```
This is still global code, you just put it under the `if` statement  🤷‍♂️
In this way, code is less reusable

SpirinEgor commented on main_summarizer.py:
```diff
@@ -0,0 +1,21 @@
+	def summarize(self, text, min_size):
+		trimmed_text = text
+
+		while len(self.pipe.tokenizer(trimmed_text).tokens()) > self.MAX_TOKEN_COUNT:
```
Tokenization is a deterministic process, you should call it once.

letit6E replied:
But I'm changing the variable trimmed_text in a loop, what do you mean by "deterministic" when the function is called with different inputs?

> Tokenization is a deterministic process, you should call it once.

SpirinEgor replied:
You can do something like that:
```python
tokens = tokenizer(text).tokens()
tokens = tokens[:self.MAX_TOKEN_COUNT]
```

SpirinEgor reviewed (approved):
A little bit strange solution, it is definitely not in a pythonic way and not optimized at all. But it is working 👍
